src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1171,6 +1200,37 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
       throw failures.failure(info, elementsShouldSatisfyExactly(actual, unsatisfiedRequirements, info));
 
   }
+  
+  /**
+   * Verifies that the actual iterable has the same size as the given consumers and 
+   * that at least one permutation of elements in the iterable exists that satisfies 
+   * the individual consumers in order.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given {@code Iterable}.
+   * @param consumers the consumers that are expected to be satisfied by the elements of the given {@code Iterable}.
+   * @throws NullPointerException if the given consumers array or any consumer is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} and consumers don't have the same size.
+   * @throws AssertionError if any {@code Consumer} in the consumers cannot be satisfied by elements in the given {@code Iterable}.
+   */
+  public <E> void assertSatisfiesExactlyInAnyOrder(AssertionInfo info, Iterable<? extends E> actual, @SuppressWarnings("unchecked") Consumer<? super E>... consumers) {
+    assertNotNull(info, actual);
+    requireNonNull(consumers, "The Consumer<? super E>... expressing the assertions consumers must not be null");
+    for (Consumer<? super E> consumer : consumers)
+      requireNonNull(consumer, "The element in consumers must not be null");
+
+    checkSizes(actual, sizeOf(actual), consumers.length, info);
+
+    List<E>[] satisfiedElementsLists = stream(consumers)
+                                                        .map(listFilteredBySatisfiedElements(actual))
+                                                        .<List<E>> toArray(List[]::new);
+
+    if (!isSatisfied(satisfiedElementsLists, 0))
+      throw failures.failure(info, shouldSatisfy(actual));
```

Comment:
we better have a  `shouldSatisfyExactlyInAnyOrder` method here that will build an error message explaining why the assertions failed.

Reply:
Yes, I will rename the method accordingly.